posydon/unit_tests/utils/test_common_functions.py

```diff
@@ -0,0 +1,2065 @@
+"""Unit tests of posydon/utils/common_functions.py
+"""
+
+__authors__ = [
+    "Matthias Kruckow <Matthias.Kruckow@unige.ch>"
+]
+
+# import the module which will be tested
+import posydon.utils.common_functions as totest
+
+# import other needed code for the tests, which is not already imported in the
+# module you like to test
+from pytest import fixture, raises, warns, approx
+from inspect import isroutine, isclass
+from scipy.interpolate import interp1d
+from posydon.binary_evol.binarystar import BinaryStar as BinaryStar_class
+from posydon.binary_evol.singlestar import SingleStar as SingleStar_class
+from posydon.utils.posydonwarning import (EvolutionWarning,\
+    InappropriateValueWarning, ApproximationWarning, InterpolationWarning,\
+    ReplaceValueWarning, ClassificationWarning)
+
+@fixture
+def BinaryStar():
+    # initialize a BinaryStar instance, which is a required argument
+    return BinaryStar_class()
+
+@fixture
+def SingleStar():
+    # initialize a SingleStar instance, which is a required argument
+    return SingleStar_class()
+
+@fixture
+def star_profile():
+    # generate a profile of a test star
+    profile = totest.np.empty((4,), dtype=[(f, 'f8') for f in ['mass', 'dm',\
+              'radius', 'log_R', 'energy', 'x_mass_fraction_H',\
+              'y_mass_fraction_He', 'z_mass_fraction_metals',\
+              'neutral_fraction_H', 'neutral_fraction_He', 'avg_charge_He']])
+    profile['mass'] = totest.np.array([1.0, 0.5, 0.1, 0.001])
+    # get dm as differences from mass and the last mass entry (like next would
+    # be 0)
+    profile['dm'][:-1] = profile['mass'][:-1] - profile['mass'][1:]
+    profile['dm'][-1] = profile['mass'][-1]
+    profile['radius'] = totest.np.array([1.0, 0.5, 0.1, 0.001])
+    # get log10 of radius
+    profile['log_R'] = totest.np.log10(profile['radius'])
+    profile['energy'] = totest.np.array([1.0, 0.5, 0.1, 0.001])*1.0e+16
+    profile['x_mass_fraction_H'] = totest.np.array([0.7, 0.5, 0.1, 0.001])
+    profile['y_mass_fraction_He'] = totest.np.array([0.2, 0.5, 0.8, 0.2])
+    # get Z=1-X-Y
+    profile['z_mass_fraction_metals'] = 1.0-profile['x_mass_fraction_H']\
```

Comment:
It's the other way around, I wrote in the style to not use spaces to save space because of the character limit on lines.

Reply:
I would suggest adding the spaces since it's pep8, which is what our posydon style suggests: 
https://peps.python.org/pep-0008/
From there:
> Always surround these binary operators with a single space on either side: assignment (=), augmented assignment (+=, -= etc.), comparisons (==, <, >, !=, <>, <=, >=, in, not in, is, is not), Booleans (and, or, not).

Reply:
I'll check through again tomorrow to add more spaces and consequently line breaks.